frontend/ui/widget/hyphenationlimits.lua

```diff
@@ -0,0 +1,280 @@
+                GestureRange:new{
+                    ges = "tap",
+                    range = Geom:new{
+                        w = self.screen_width,
+                        h = self.screen_height,
+                    }
+                },
+            },
+         }
+    end
+    self:update()
+end
+
+function HyphenationLimitsWidget:update()
+    -- This picker_update_callback will be redefined later. It is needed
+    -- so we can have our MovableContainer repainted on NumberPickerWidgets
+    -- update It is needed if we have enabled transparency on MovableContainer,
+    -- otherwise the NumberPicker area gets opaque on update.
+    local picker_update_callback = function() end
+    local left_widget = NumberPickerWidget:new{
+        show_parent = self,
+        width = self.picker_width,
+        value = self.left_value,
+        value_min = self.left_min,
+        value_max = self.left_max,
+        wrap = false,
+        update_callback = function() picker_update_callback() end,
+    }
+    local right_widget = NumberPickerWidget:new{
+        show_parent = self,
+        width = self.picker_width,
+        value = self.right_value,
+        value_min = self.right_min,
+        value_max = self.right_max,
+        wrap = false,
+        update_callback = function() picker_update_callback() end,
+    }
+    local hyph_group = HorizontalGroup:new{
+        align = "center",
+        VerticalGroup:new{
+            align = "center",
+            VerticalSpan:new{ width = Size.span.vertical_large },
+            TextBoxWidget:new{
+                text = _("Left"),
+                alignment = "center",
+                face = self.title_face,
+                width = self.picker_width,
+            },
+            left_widget,
+        },
+        VerticalGroup:new{
+            align = "center",
+            VerticalSpan:new{ width = Size.span.vertical_large },
+            TextBoxWidget:new{
+                text = _("Right"),
+                alignment = "center",
+                face = self.title_face,
+                width = self.picker_width,
+            },
+            right_widget,
+        },
+    }
+
+    local hyph_title = FrameContainer:new{
+        padding = Size.padding.default,
+        margin = Size.margin.title,
+        bordersize = 0,
+        TextWidget:new{
+            text = self.title_text,
+            face = self.title_face,
+            bold = true,
+            width = self.width,
+        },
+    }
+    local hyph_line = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Size.line.thick,
+        }
+    }
+    local hyph_bar = OverlapGroup:new{
+        dimen = {
+            w = self.width,
+            h = hyph_title:getSize().h
+        },
+        hyph_title,
+        CloseButton:new{ window = self, padding_top = Size.margin.title, },
+    }
+
+    local hyph_into_text = _("Set the minimum length of word accepted on both side of the hyphenation.")
```

Comment:
both sides

Although I'd say there's something slightly off about the whole phrase.

`Set minimum length before hyphenation occurs`, perhaps?